presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java

```diff
@@ -106,6 +106,15 @@ public void testRcTextCharDecoding()
         testRcTextCharDecoding(true);
     }
 
+    @Test
+    public void testInvalidPartitionValue()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE invalid_partition_value (a int, b varchar) WITH (partitioned_by = ARRAY['b'])");
+        assertQueryFails("INSERT INTO invalid_partition_value VALUES (4, '\n')", "\"\n\" contains characters not in the range of 0x20 - 0x7E");
+        assertUpdate("DROP TABLE invalid_partition_value");
```

Comment:
You can do 'create table as select' instead. This way, you won't need drop. And you will only need one query.

Alternatively, you could also test both CTAS and INSERT. It's up to you.